### PR TITLE
docs: add jsdocs for all properties

### PR DESCRIPTION
**Pull request by haxxmaxx**
- Adding the `spec` command to generate api specs
- Add all properties to `object` properties

### CODE REVIEW

File: spec-configs/props.conf.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  package: path.resolve(__dirname, '../package.json'),
+  api: {
+    stability: 'experimental',
+    properties: {
+      'x-qlik-visibility': 'public',
+    },
+    visibility: 'public',
+    name: `${pkg.name}:properties`,
+    version: pkg.version,
+    description: 'Table generic object definition',
```
Comment (cbt1) [inline spec-configs/props.conf.js:16]: ☝️

File: src/ext.js [hunk 2]
```diff
@@ -162,8 +162,9 @@ export default function ext({ translator, automationsEnabled }) {
                   ref: 'automation',
                   dropdownOnly: true,
                   options: async () => {
-                    const automations = await fetch('../api/v1/items?resourceType=automation&limit=100').then((response) =>
-                      response.json());
+                    const automations = await fetch('../api/v1/items?resourceType=automation&limit=100').then(
```
Comment (cbt1) [inline src/ext.js:165]: Nitpicking here but I think a common praxis is to line break on new method calls.

```js
const automations = await fetch('../api/v1/items?resourceType=automation&limit=100')
  .then((response) => response.json());
```

File: src/ext.js [hunk 3]
```diff
@@ -260,6 +261,7 @@ export default function ext({ translator, automationsEnabled }) {
               component: 'switch',
               translation: 'properties.enableToggle',
               ref: 'useEnabledCondition',
+              defaultValue: false,
```
Comment (cbt1) [inline src/ext.js:264]: Is there a risk that these default values have changed now that they are explicitly set?

File: api-specifications/properties.json [hunk 4]
```diff
@@ -0,0 +1,625 @@
+{
+  "scriptappy": "1.0.0",
+  "info": {
+    "name": "@nebula.js/sn-action-button:properties",
+    "description": "Table generic object definition",
```
Comment (cbt1) [inline api-specifications/properties.json:5]: ☝️

File: api-specifications/properties.json [hunk 5]
```diff
@@ -0,0 +1,625 @@
...
+    "version": "1.14.3",
+    "license": "MIT",
+    "stability": "experimental",
```
Comment (cbt1) [inline api-specifications/properties.json:8]: Is this still experimental?

**Comment by cbt1** [comment]
Ideally this PR should be split into a `docs` and `refactor` PR. As you are now making code changes and updating the docs, so your `docs` tag is not really valid.

File: src/ext.js (hunk 3 above)
Comment (haxxmaxx) [reply src/ext.js:264]: well yes, if I did it incorrectly. I can double-check, maybe even do some manual sanity checks

File: src/ext.js (hunk 2 above)
Comment (haxxmaxx) [reply src/ext.js:165]: sure, lint apparently changed this

**Comment by haxxmaxx** [comment]
> Ideally this PR should be split into a `docs` and `refactor` PR. As you are now making code changes and updating the docs, so your `docs` tag is not really valid.

Hmm good point. I guess I could start with the code changes

File: api-specifications/properties.json (hunk 5 above)
Comment (haxxmaxx) [reply api-specifications/properties.json:8]: Not sure, when does it stop being experimental?

File: src/object-properties.js [hunk 6]
```diff
@@ -1,38 +1,62 @@
...
+
 const properties = {
   /**
-   * @type {boolean}
+   * Current version of this generic object definition.
+   * @type {string}
+   * @default
    */
-  showTitles: false,
+  version: (process && process.env && process.env.PACKAGE_VERSION) || '1.0.0',
```
Comment (cbt1) [inline src/object-properties.js:14]: When is the version nbr not available?

Comment (haxxmaxx) [reply src/object-properties.js:14]: When running rendering tests, but it is still failing. process comes from node it seems, I'll have to get around it in a different way

**Review by cbt1 — APPROVED** [review]
LGTM, I guess you need to figure out why the build is failing

**Pull request merged**